examples/connectivity/plot_extract_regions_canica_maps.py

```diff
@@ -0,0 +1,94 @@
+"""
+Regions extraction using Canonical ICA maps
+===========================================
+
+This example specifically shows how to segment each ICA map (a 4D Nifti
+image/object) into a distinct seperated brain region and extracts timeseries
+signals from each seperated region. Both can be done at the same time
+using `RegionExtractor`.
+
+We used 40 resting state functional datasets and `CanICA` for 4D components.
+
+This example motivates, how to use `RegionExtractor` to study
+functional connectomes using correlation and partial correlation
+matrices.
+Please see the related documentation of `RegionExtractor` for more details.
+"""
+import numpy as np
+from nilearn import datasets
+print " -- Fetching ADHD resting state functional datasets -- "
+adhd_dataset = datasets.fetch_adhd(n_subjects=40)
+func_filenames = adhd_dataset.func
+confounds = adhd_dataset.confounds
+
+from nilearn.input_data import NiftiMasker
+print " -- Computing the mask from the data-- "
+func_filename = func_filenames[0]
+masker = NiftiMasker(standardize=False, mask_strategy='epi')
+masker.fit(func_filename)
+mask_img = masker.mask_img_
+
+from nilearn.decomposition.canica import CanICA
+print " -- Canonical ICA decomposition of functional datasets -- "
+# Initialize canica parameters
+n_components = 20
+canica = CanICA(n_components=n_components, smoothing_fwhm=6.,
+                memory="nilearn_cache", memory_level=5,
+                threshold=3., random_state=0)
+
+canica.fit(func_filenames)
+components_img = canica.masker_.inverse_transform(canica.components_)
+
+# Segementation step: Region extraction from ICA maps
+# Signals step: Average timeseries signal extraction
+# Both are done by calling fit_transform()
+from nilearn.regions import region_extractor
+print " -- Extracting regions from ICA maps and timeseries signals -- "
+reg_ext = region_extractor.RegionExtractor(components_img,
```

Comment:
`reg_ext` makes me think of regular expressions. Maybe `extractor` is sufficient ?